hydromt_sfincs/utils.py

```diff
@@ -1184,3 +1185,77 @@ def _downscale_floodmap_da(
         hmax = hmax.where(mask)
 
     return hmax
+
+
+def find_uv_indices(mask: xr.DataArray):
```

Comment:
I'm not sure what indices are returned exactly. It would be good to add what the outputs represent and what types these are in the method.

Reply:
Changed docstring into this, hopefully that clarifies your question?:

```
    """The subgrid tables for a regular SFINCS grid are organized as flattened arrays, meaning 
    2D arrays (y,x) are transformed into 1D arrays, only containing values for active cells.

    For the cell centers, this is straightforward, we just find the indices of the active cells.
    However, the u and v points are saved in combined arrays. Since u and v points are absent
    at the boundaries of the domain, the index arrays are used to determine the location of the
    u and v points in the combined flattened arrays.


    
    Parameters
    ----------
    mask: xr.DataArray
        Mask with integer values specifying the active cells of the SFINCS domain.

    Returns
    -------
    index_nm: np.ndarray
        Index array for the active cell centers.
    index_mu1: np.ndarray
        Index of upstream u-point in combined uv-array.
    index_nu1: np.ndarray
        Index of upstream v-point in combined uv-array.

    """
```

Reply:
Thanks! This give more context to read the code. I asked because I think that it might be easier (and faster as it saves many python loops) to get the 2D indices fashion by stacking the mask and shifted masks for u and v to find all valid indices. But I don't have sufficiently thorough understanding of the method to refactor it.

Reply:
Yes, agree on this, but I couldn't further optimize this.

Think the main problem here is that there are no u- and/or v point at the outer edges of the domain, which results sometimes in two u-points in a row and sometimes two v-points in a row, whereas the usually are alternating (in the combined uv matrix)